src/js/letters/actions/letters.js

```diff
@@ -76,35 +75,37 @@ export function getAddressFailure() {
 
 export function getMailingAddress() {
   return (dispatch) => {
-    apiRequest(
+    return apiRequest(
       '/v0/address',
       null,
-      response => {
+      // on fetch success
+      (response) => {
         const responseCopy = Object.assign({}, response);
-        const address = Object.assign({}, response.data.attributes.address);
-        // Translate military-only fields into generic ones; we'll translate them back later if necessary
-        if (address.type === ADDRESS_TYPES.military) {
-          address.city = address.militaryPostOfficeTypeCode;
-          address.stateCode = address.militaryStateCode;
-          address.countryName = 'USA';
-          delete address.militaryPostOfficeTypeCode;
-          delete address.militaryStateCode;
+        const addressCopy = Object.assign({}, response.data.attributes.address);
```

Comment:
renaming address to addressCopy makes this whole block look new, but that's all that's going on here.